src/api-calls.js

```diff
@@ -1,150 +1,32 @@
-  
-//         getSingleTraveler(travelerId)
-//           .then((travelerData) => {
-//             if (travelerData) {
-//               welcomeMessage.textContent = `Welcome, ${travelerData.name}!`;
-//               console.log(travelerData);
-//             } else {
-//               alert("Traveler not found. Please try again.");
-//             }
-//           })
-//           .catch((error) => {
-//             console.error(error);
-//           });
-  
-//         loginForm.style.display = "none";
-//       } else {
-//         alert("Invalid login credentials. Please try again.");
-//       }
-//     });
-// });
-// console.log('This is the JavaScript entry file - your code begins here.');
-// getTravellers()
-
-// window.addEventListener("load", () =>{
-//   Promise.all([getTravellers(),()=>getSingleTraveler(travelerId),getAllTrips(),getAllDestinations()])
-//   .then((data) =>{
-//       travelers = data[0]
-//       singleTravler = data[1]()
-//       trips = data[2]
-//       destinations = data[3]
-//   })
-// })
-
-// function displayPendingTrips() {
-//   // Filter trips with a status of "pending"
-// const pendingTrips = trips.filter(trip => trip.status === 'pending');
-
-// if (pendingTrips.length === 0) {
-//     // No pending trips found
-//   pendingTripsSection.innerHTML = '<p>No pending trips at the moment.</p>';
-// } else {
-//     // Create and append HTML elements for each pending trip
-//   const tripList = document.createElement('ul');
-//   tripList.classList.add('trip-list');
-
-//   pendingTrips.forEach(trip => {
-//   const tripItem = document.createElement('li');
-//   tripItem.innerHTML = `
-//       <h3>Trip to ${destinations.find(dest => dest.id === trip.destinationID).destination}</h3>
-//       <p>Date: ${trip.date}</p>
-//       <p>Duration: ${trip.duration} days</p>
-//       <p>Number of Travelers: ${trip.travelers}</p>
-//   `;
-
-//   tripList.appendChild(tripItem);
-//   });
-
-//   pendingTripsSection.appendChild(tripList);
-// }
-// }
+export function fetchTravelers() {
+  return fetch("http://localhost:3001/api/v1/travelers")
+      .then((response) => response.json())
+      .then((data) => data.travelers);
+}
+
+export function fetchTrips() {
+  return fetch("http://localhost:3001/api/v1/trips")
+      .then((response) => response.json())
+      .then((data) => data.trips);
+}
+
+export function fetchDestinations() {
+  return fetch("http://localhost:3001/api/v1/destinations")
+      .then((response) => response.json())
+      .then((data) => data.destinations);
+}
+
+export function postNewTrip(newTrip) {
+  return fetch("http://localhost:3001/api/v1/trips", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(newTrip),
+  })
+    .then((response) => response.json())
+    .then(() => {
+      // After posting the new trip, fetch the updated list of trips
+      return fetchTrips();
+    });
+}
```

Comment:
consider exporting functions as one block of exports vs individually exporting each one for future projects/cross-file clarity